lib/BoundaryValueDiffEqMIRKN/test/mirkn_basic_tests.jl

```diff
@@ -0,0 +1,116 @@
+    end
+
+    function bc!(res, du, u, p, t)
+        res[1] = u[1][1]
+        res[2] = u[end][1]
+        res[3] = u[1][3] + 1
+        res[4] = u[end][3] - 1
+        res[5] = du[1][1]
+        res[6] = du[end][1]
+    end
+
+    function bc(du, u, p, t)
+        return [u[1][1], u[end][1], u[1][3] + 1, u[end][3] - 1, du[1][1], du[end][1]]
+    end
+    function bca!(resa, du, u, p)
+        resa[1] = u[1]
+        resa[2] = u[3] + 1
+        resa[3] = du[1]
+    end
+    function bcb!(resb, du, u, p)
+        resb[1] = u[1]
+        resb[2] = u[3] - 1
+        resb[3] = du[1]
+    end
+
+    function bca(du, u, p)
+        [u[1], u[3] + 1, du[1]]
+    end
+    function bcb(du, u, p)
+        [u[1], u[3] - 1, du[1]]
+    end
+
+    u0 = [1.0, 1.0, 1.0]
+    tspan = (0.0, 1.0)
+
+    probArr = [SecondOrderBVProblem(test!, bc!, u0, tspan),
+        SecondOrderBVProblem(test, bc, u0, tspan),
+        TwoPointSecondOrderBVProblem(
+            test!, (bca!, bcb!), u0, tspan, bcresid_prototype = (zeros(3), zeros(3))),
+        TwoPointSecondOrderBVProblem(
+            test, (bca, bcb), u0, tspan, bcresid_prototype = (zeros(3), zeros(3)))]
+
+    @testset "MIRKN$order" for order in (4, 6)
+        @testset "Problem $i" for i in 1:4
+            sol = solve(probArr[i], mirkn_solver(Val(order)); dt = 0.01)
+            @test SciMLBase.successful_retcode(sol)
+        end
+    end
+end
+#=
+@testitem "Convergence on Linear" begin
+    using LinearAlgebra, DiffEqDevTools
+
+    function f!(ddu, du, u, p, t)
+        ddu[1] = u[1]
+    end
+    function f(du, u, p, t)
+        return u[1]
+    end
+    function bc!(res, du, u, p, t)
+        res[1] = u[1][1] - 1
+        res[2] = u[end][1]
+    end
+    function bc(du, u, p, t)
+        return [u[1][1] - 1,
+            u[end][1]]
+    end
+    function bc_a!(res, du, u, p)
+        res[1] = u[1] - 1
+    end
+    function bc_b!(res, du, u, p)
+        res[1] = u[1]
+    end
+    function bc_a(du, u, p)
+        return [u[1] - 1]
+    end
+    function bc_b(du, u, p)
+        return [u[1]]
+    end
+    u0 = [1.0]
+    tspan = (0.0, 1.0)
+    odef1 = ODEFunction(f!,
+    analytic = (du, u0, p, t) -> [(exp(-t) - exp(t - 2)) / (1 - exp(-2)),
+        (-exp(-t) - exp(t - 2)) / (1 - exp(-2))])
+    odef2 = ODEFunction(f,
+    analytic = (u0, p, t) -> [(exp(-t) - exp(t - 2)) / (1 - exp(-2)),
+        (-exp(-t) - exp(t - 2)) / (1 - exp(-2))])
+    probArr = [SecondOrderBVProblem(odef1, bc!, u0, tspan),
+        SecondOrderBVProblem(odef2, bc, u0, tspan),
+        TwoPointSecondOrderBVProblem(odef1, (bc_a!, bc_b!), u0, tspan),
+        TwoPointSecondOrderBVProblem(odef2, (bc_a, bc_b), u0, tspan)]
+    dts = 1 .// 2 .^ (3:-1:1)
+    @testset "MIRKN$order" for order in (4, 6)
+        sim = test_convergence(
+                    dts, prob, mirkn_solver; abstol = 1e-8, reltol = 1e-8)
+        @test sim.errors[:final]≈order atol=testTol
+    end
+end
```

Comment:
The convergence tests for MIRKN4 and MIRK6 pass locally. Since the `ODEFunction(f, analytic = analytic_solution)` has arguments number checkings for `f`, how can we construct such a function with an analytic solution for `SecondOrderBVProblem`? I am thinking maybe adding a new function type to allow this in SciMLBase.jl?

Reply:
Nothing to worry about then, just forgot that I already added `DynamicalBVPFunction` a few months ago😅